pkg/agent/config_test.go

```diff
@@ -25,14 +25,30 @@ func Test_ValidateAndCombineConfig(t *testing.T) {
 	// OAuth mode.
 	fakeCredsPath := withFile(t, `{"user_id":"foo","user_secret":"bar","client_id": "baz","client_secret": "foobar","auth_server_domain":"bazbar"}`)
 
-	t.Run("period must be given with either --period/-p or period field in config", func(t *testing.T) {
+	// Usually, the namespace is guessed from the file
+	// /var/run/secrets/kubernetes.io/serviceaccount/namespace. But since we
+	// can't realistically set that file in tests, we pass the flag
+	// --install-namespace in all the tests.
```

Comment:
I added `POD_NAMESPACE` to deployment.yaml.
So perhaps instead of getting the namespace from the file it could get it from the environment.
Then maybe  you could just add a `t.SetEnv("POD_NAMESPACE", "venafi") ` to these tests and avoid all the churn in the  test code below?

Reply:
> I added `POD_NAMESPACE` to deployment.yaml. So perhaps instead of getting the namespace from the file it could get it from the environment.

Good point, it seems like a better idea. This file-based discovery of the namespace was implicit, and I'd very much prefer clearly see what is being passed to the process in the deployment, like what you did with `POD_NAMESPACE`, `POD_NAME`, and `POD_UID`.

That said, I've already implemented the discovery of the namespace using the file-based approach in the Venafi Connection auth mode. I guess I should migrate that to using `POD_NAMESPACE` instead too. I'll do that in a follow-up PR, though.

Reply:
I thought you'd be able to use `t.SetEnv` in the parent test scope and have it applied to all the sub-tests.
If that's not possible then perhaps my POD_NAMESPACE suggestion was misleading.
I'm honestly not sure what is the best option.

Reply:
You are correct, I could have used `t.SetEnv` on the top level `t`.

The reason I haven't done so is that I am somewhat reluctant doing that because it would hide (by hide, I mean it wouldn't be obvious) an important input to these test cases.

Thinking about it, I guess it isn't that big of a deal if I set it on the top level `t`...